cmd/transflect-operator/operator.go

```diff
@@ -289,7 +290,10 @@ func (o *operator) processFilter(rs *appsv1.ReplicaSet) error {
 	port := grpcPort(rs)
 	if port == 0 {
 		if err := o.deleteFilter(context.Background(), rs); err != nil {
-			// TODO fix for is not exist, remove from active state and deployment locker.
+			if k8errors.IsNotFound(err) {
+				o.activeState.Delete(deployKey)
+				o.deploymentLocker.delete(deployKey)
+			}
```

Comment:
this will fall though and return an error. I would not do that. I would just accept it was already deleted and move on. In that case, you could re-use the existing code below by inverting this:
```
if !k8errors.IsNotFound(err) {
    return err
}
```